drivers/flash/flash_stm32_ospi.c

```diff
@@ -887,8 +898,7 @@ static int flash_stm32_ospi_read(const struct device *dev, off_t addr,
 	}
 
 	/* Instruction and DummyCycles are set below */
-	cmd.Address = addr;
-	cmd.AddressSize = stm32_ospi_hal_address_size(dev);
```

Comment:
Quad and dual flashes might not work with a 32-bits address size.
`ospi_prepare_cmd` enforces 32-bits address size. Please use 24-bits for non `OSPI_OPI_MODE`.

Reply:
ok